### PR TITLE
Makes it so beans can't hit dead simple animals anymore.

### LINKED ISSUE #7342: Ravager can slash dead simplemobs for free fury
### Description

Read title

### Test Merges

#7270 #7099 #7194 #7257 #7164 #7181

### Reproduction Steps

1. Get dead simplemob
2. Slash it as rav
3. 100% Fury bar

### Screenshots

https://user-images.githubusercontent.com/24631139/125176738-7413c300-e208-11eb-9b8a-a3ba6f47c77c.mov

### PR DESCRIPTION
**Pull request by lbnesquik**
## About The Pull Request
I could make it a check for ravagers only but whatever.
This just makes it impossible to slash dead simple animals.

## Why It's Good For The Game
Fixes #7342

## Changelog
:cl:
fix: fixes ravager getting free rage from hitting simple mobs.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/attack_alien.dm [hunk 1]
```diff
@@ -73,6 +73,10 @@
 	if(!can_xeno_slash(X))
 		return FALSE
 
+	if(isanimal(src) && (stat == DEAD))
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/attack_alien.dm:76]: bad. In general, istype(src) are verboten. 

Here you must overide attack_alien_harm for animal mobs.

**Review by BraveMole — CHANGES_REQUESTED** [review]
I'm not a big fan on this, when the issue is only ravager fury. Can you check if the mob attacked is dead, and if so not give fury?

**Comment by lbnesquik** [comment]
Fuck you bravemole, making me do shit

**Comment by BraveMole** [comment]
Kek self resolving an issue that will be blocking, bad smalls

File: code/datums/elements/plasma_on_attack.dm [hunk 2]
```diff
@@ -20,5 +20,8 @@
 /datum/element/plasma_on_attack/proc/damage_dealt(datum/source, mob/living/attacked, damage)
 	SIGNAL_HANDLER
 	var/mob/living/carbon/xenomorph/furious = source
+	if(attacked.stat == DEAD)
+		to_chat(furious, span_notice("This thing is already dead. Why bother touching it?"))
+		return
 	furious.gain_plasma(damage * damage_plasma_multiplier)
```
Comment (TiviPlus) [inline code/datums/elements/plasma_on_attack.dm:26]:
```suggestion
		return NONE
	furious.gain_plasma(damage * damage_plasma_multiplier)
```

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**